packages/mdc-tab-bar/foundation.js

```diff
@@ -0,0 +1,365 @@
+     * out of view to the right. From there, we either increment or decrement
+     * the index.
+     */
+    const relativeRootLeft = tabDimensions.rootLeft - scrollPosition;
+    const relativeRootRight = tabDimensions.rootRight - scrollPosition - barWidth;
+    const relativeRootDelta = relativeRootLeft + relativeRootRight;
+    const leftEdgeIsCloser = relativeRootLeft < 0 || relativeRootDelta < 0;
+    const rightEdgeIsCloser = relativeRootRight > 0 || relativeRootDelta > 0;
+
+    if (leftEdgeIsCloser) {
+      return index - 1;
+    }
+
+    if (rightEdgeIsCloser) {
+      return index + 1;
+    }
+
+    return -1;
+  }
+
+  /**
+   * Determines the index of the next adjacent tab in RTL
+   * @param {number} index The index of the tab
+   * @param {!MDCTabDimensions} tabDimensions Whether the next tab is left or right
+   * @param {number} scrollDistance The distance scrolled
+   * @param {number} scrollWidth The width of the scroller
+   * @param {number} barWidth The width of the bar
+   * @return {number}
+   * @private
+   */
+  findTabIndexClosestToEdgeRTL_(index, tabDimensions, scrollDistance, scrollWidth, barWidth) {
+    const rootLeft = scrollWidth - tabDimensions.rootLeft - barWidth - scrollDistance;
+    const rootRight = scrollWidth - tabDimensions.rootRight - scrollDistance;
+    const rootDelta = rootLeft + rootRight;
+    const leftEdgeIsCloser = rootLeft > 0 || rootDelta > 0;
+    const rightEdgeIsCloser = rootRight < 0 || rootDelta < 0;
+
+    if (leftEdgeIsCloser) {
+      return index + 1;
+    }
+
+    if (rightEdgeIsCloser) {
+      return index - 1;
+    }
+
+    return -1;
+  }
+
+  /**
+   * Returns whether a given index is inclusively between the ends
+   * @param {number} index The index to test
+   * @private
+   */
+  indexIsInRange_(index) {
+    return index >= 0 && index < this.adapter_.getTabListLength();
+  }
+
+  /**
+   * Returns the view's RTL property
+   * @return {boolean}
+   * @private
+   */
+  isRTL_() {
+    return this.adapter_.isRTL();
+  }
+
+  /**
+   * Scrolls the tab at the given index into view for left-to-right useragents
+   * @param {number} index The index of the tab to scroll into view
+   * @private
+   */
+  scrollIntoView_(index) {
+    const scrollPosition = this.adapter_.getScrollPosition();
+    const barWidth = this.adapter_.getOffsetWidth();
+    const tabDimensions = this.adapter_.getTabDimensionsAtIndex(index);
+    const nextIndex = this.findTabIndexClosestToEdge_(index, tabDimensions, scrollPosition, barWidth);
+
+    if (!this.indexIsInRange_(nextIndex)) {
+      return;
+    }
+
+    const scrollIncrement = this.calculateScrollIncrement_(index, nextIndex, scrollPosition, barWidth);
+    this.adapter_.incrementScroll(scrollIncrement);
+  }
+
+  /**
+   * Scrolls the tab at the given index into view in RTL
+   * @param {number} index The tab index to make visible
+   * @private
+   */
+  scrollIntoViewRTL_(index) {
+    const barWidth = this.adapter_.getOffsetWidth();
```

Comment:
Nit: sync these up with the order of variables in the non-RTL method to make it easier to see the actual differences

Reply:
Ah my bad, I misunderstood. I'll update.